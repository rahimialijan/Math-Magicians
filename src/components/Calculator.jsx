import Buttons from './Buttons';

const Calculator = () => <Buttons />;
export default Calculator;
