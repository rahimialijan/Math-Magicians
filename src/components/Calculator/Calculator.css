.calc-container {
  display: flex;
  height: 65%;
  width: 60%;
  justify-content: space-between;
  margin-top: 50px;
}

.wrapper {
  display: grid;
  height: 100%;
  width: 50%;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(6, 1fr);
  align-self: center;
  background-color: rgb(134, 132, 132);
  border-radius: 1rem;
  -webkit-box-shadow: 7px 7px 5px 0 rgba(50, 50, 50, 0.75);
  -moz-box-shadow: 7px 7px 5px 0 rgba(50, 50, 50, 0.75);
  box-shadow: 7px 7px 5px 0 rgba(50, 50, 50, 0.75);
}

.screen {
  grid-column: 1/-1;
  color: white;
  text-align: end;
  padding-right: 10px;
  padding-top: 18px;
  font-size: 2rem;
  overflow: hidden;
  background-color: rgba(134, 132, 132, 0.703);
  border: 2px solid rgb(101, 99, 99);
  border-top-right-radius: 1rem;
  border-top-left-radius: 1rem;
}

.btn {
  display: grid;
  place-content: center;
  background-color: rgb(220, 220, 220);
  color: black;
  cursor: pointer;
  border-radius: 4px;
  font-weight: bold;
  border: 1px solid rgb(185, 185, 185);
  font-size: 1.3rem;
}

.btn:hover {
  background-color: gray;
}

.light-gray {
  background-color: rgb(220, 220, 220);
  color: black;
}

.light-gray:hover {
  background-color: white;
}

.orange {
  background-color: rgba(255, 166, 0, 0.842);
}

.orange:hover {
  background-color: white;
  color: orange;
}

.zero {
  grid-column: 1/span 2;
  place-content: unset;
  align-items: center;
  padding-left: 2rem;
}
