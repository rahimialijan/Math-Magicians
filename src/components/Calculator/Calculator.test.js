import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from './Calculator';

describe('Calculator', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('handles button clicks correctly', () => {
    const { getByText, getByTestId } = render(<Calculator />);

    fireEvent.click(getByText('7'));
    expect(getByTestId('screen')).toHaveTextContent('7');

    fireEvent.click(getByText('+'));
    expect(getByTestId('screen')).toHaveTextContent('7');

    fireEvent.click(getByText('3'));
    expect(getByTestId('screen')).toHaveTextContent('3');

    fireEvent.click(getByText('='));
    expect(getByTestId('screen')).toHaveTextContent('10');
  });
});
