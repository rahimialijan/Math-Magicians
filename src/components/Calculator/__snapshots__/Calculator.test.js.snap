// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<DocumentFragment>
  <div
    class="calc-container"
  >
    <h1>
      Let's do some Math!
    </h1>
    <div
      class="wrapper"
    >
      <div
        class="screen"
        data-testid="screen"
      >
        0
      </div>
      <button
        class="btn light-gray"
        type="button"
      >
        AC
      </button>
      <button
        class="btn light-gray"
        type="button"
      >
        +/-
      </button>
      <button
        class="btn light-gray"
        type="button"
      >
        %
      </button>
      <button
        class="btn orange"
        type="button"
      >
        รท
      </button>
      <button
        class="btn"
        type="button"
      >
        7
      </button>
      <button
        class="btn"
        type="button"
      >
        8
      </button>
      <button
        class="btn"
        type="button"
      >
        9
      </button>
      <button
        class="btn orange"
        type="button"
      >
        x
      </button>
      <button
        class="btn"
        type="button"
      >
        4
      </button>
      <button
        class="btn"
        type="button"
      >
        5
      </button>
      <button
        class="btn"
        type="button"
      >
        6
      </button>
      <button
        class="btn orange"
        type="button"
      >
        +
      </button>
      <button
        class="btn"
        type="button"
      >
        1
      </button>
      <button
        class="btn"
        type="button"
      >
        2
      </button>
      <button
        class="btn"
        type="button"
      >
        3
      </button>
      <button
        class="btn orange"
        type="button"
      >
        -
      </button>
      <button
        class="btn zero"
        type="button"
      >
        0
      </button>
      <button
        class="btn"
        type="button"
      >
        .
      </button>
      <button
        class="btn orange"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
